compiler/src/dotty/tools/io/VirtualFile.scala

```diff
@@ -40,15 +40,32 @@ class VirtualFile(val name: String, override val path: String) extends AbstractF
     this.content = content
   }
 
+  /**
+    * Initializes this instance with the specified path
+    * and a name taken from the last path element.
+    *
+    * @param path the path of the virtual file to be created
+    * @param content the initial contents of the virtual file
+    * @return     the created virtual file
+    */
+  def this(path: JPath, content: Array[Byte]) = {
+    this(path.toString.replace(java.io.File.separatorChar, '/'), content)
```

Comment:
```suggestion
    this(jpath.getFileName().toString(), path)
    this.content = content
```
wouldn't make sense? I think / is only used to get the last element